pkg/ingester/transfer.go

```diff
@@ -214,6 +219,20 @@ func (i *Ingester) TransferTSDB(stream client.Ingester_TransferTSDBServer) error
 	fromIngesterID := ""
 
 	xfer := func() error {
+
+		// Validate the final directory is empty, if it exists and is empty delete it so a move can succeed
+		err := removeEmptyDir(i.cfg.TSDBConfig.Dir)
+		if err != nil {
+			return errors.Wrap(err, "Non-empty tsdb directory")
```

Comment:
I think, "remove existing TSDB directory" might be appropriate, as there could be several reasons for the error, not just non empty.

Reply:
Good call; changed.